### DISCUSSION TITLE
Close Modal Triggers 500 Error

### DISCUSSION THREAD

**Discussion by mojowill** [wire-elements/modal#363] [Q&A]
I'm upgrading to Livewire 3 and thought I had everything sorted. I can open modals without any issues however when closing them (either via a button or simply clicking away or hitting escape) I get an error rendered:

`Only arrays and Traversables can be unpacked`

I did see there was a PR about this for destroying modals and properly passing arguments. As this is closing a modal there doesn't appear to be anything I can do directly?

**Comment by PhiloNL** [comment]
Do you emit events by any chance when closing the modal?

**Reply by mojowill** [reply]
Not that we would have added in all cases when the modal closes we're done with it.

**Comment by mojowill** [comment]
Seems we do have a listener for the modal (we were having an issue with modals sending events after the user session had expired, well I seem to remember that is why this was added?)

```php
<?php

namespace App\Livewire;

use LivewireUI\Modal\ModalComponent as LivewireUIModalComponent;

class ModalComponent extends LivewireUIModalComponent
{
    public function boot(): void
    {
        $this->listeners['modalClosed'] = 'checkUser';
    }

    public function checkUser()
    {
        if (! auth()->check()) {
            exit;
        }
    }
}
```

**Comment by PhiloNL** [comment]
The error `Only arrays and Traversables can be unpacked` occurs when dispatching events using the old syntax. E.g. `['foo' => 'bar'` which is now `foo: 'bar'` (named arguments). So it seems, somewhere, it is using this old syntax. I haven't heard anyone about this before so my assumption is that this is related to your project.
